File: backend/spring-boot-kotlin/src/main/java/org/bootstrapbugz/api/shared/validator/impl/UsernameOrEmailImpl.java
package org.bootstrapbugz.api.shared.validator.impl;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;
import java.util.regex.Pattern;
import org.bootstrapbugz.api.shared.constants.Regex;
import org.bootstrapbugz.api.shared.validator.UsernameOrEmail;

public class UsernameOrEmailImpl implements ConstraintValidator<UsernameOrEmail, String> {
  public boolean isValid(String usernameOrEmail, ConstraintValidatorContext context) {
    return Pattern.compile(Regex.USERNAME).matcher(usernameOrEmail).matches()
        || Pattern.compile(Regex.EMAIL).matcher(usernameOrEmail).matches();
  }
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/shared/validator/impl/UsernameOrEmailImpl.kt
package org.bootstrapbugz.api.shared.validator.impl

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext
import org.bootstrapbugz.api.shared.constants.Regex
import org.bootstrapbugz.api.shared.validator.UsernameOrEmail
import java.util.regex.Pattern

class UsernameOrEmailImpl : ConstraintValidator<UsernameOrEmail?, String> {
    override fun isValid(usernameOrEmail: String?, context: ConstraintValidatorContext): Boolean {
        if (usernameOrEmail == null) return false
        return Pattern.compile(Regex.USERNAME).matcher(usernameOrEmail).matches() ||
                Pattern.compile(Regex.EMAIL).matcher(usernameOrEmail).matches()
    }}
